system/ee/ExpressionEngine/Addons/search/mod.search.php

```diff
@@ -1068,12 +1072,21 @@ public function search_results()
         // Fetch the cached search query
         $query = ee()->db->get_where('search', array('search_id' => $search_id));
 
-        if ($query->num_rows() == 0 or $query->row('total_results') == 0) {
-            // This should be impossible as we already know there are results
+        if ($query->num_rows() == 0) {
             return ee()->output->show_user_error(
                 'general',
                 array(lang('invalid_action'))
             );
+        } elseif ($query->row('total_results') == 0) {
+            // this works if we use the same template for results and no results
+            if ($query->row('result_page') === $query->row('no_result_page')) {
```

Comment:
The `no_result_page` column does not exist, so can't be used here. Did you mean `$this->_meta['no_result_page']`?